src/xcode/ENA/ENA/Source/Services/Risk/Provider/Helper/SoftRateLimit.swift

```diff
@@ -0,0 +1,85 @@
+////
+// 🦠 Corona-Warn-App
+//
+
+import Foundation
+import ExposureNotification
+
+
+class SoftRateLimit {
+	// MARK: - Init
+
+	init(
+		store: Store,
+		logger: Logging? = nil
+	) {
+		self.store = store
+		self.logger = logger
+	}
+
+	func setup(configuration: RiskProvidingConfiguration) -> Bool {
+		let enoughTimeHasPassed = configuration.shouldPerformExposureDetection(
+			lastExposureDetectionDate: store.exposureDetectionDate, context: " for soft rate limit"
+		)
+		let blocking = !enoughTimeHasPassed
+		if blocking {
+			Log.info("Soft rate limit is stricter than effective rate limit", log: .riskDetection, logger: logger)
+		} else {
+			Log.debug("Soft rate limit is in synch with effective rate limit", log: .riskDetection, logger: logger)
+		}
+		return blocking
+	}
+
+	func assess(
+		result: Result<[ENExposureWindow], ExposureDetection.DidEndPrematurelyReason>,
+		blocking: Bool
+	) {
+		switch result {
+		case .success:
+			if blocking {
+				Log.warning("Soft rate limit is too strict - it would have blocked this successfull exposure detection", log: .riskDetection, logger: logger)
```

Comment:
```suggestion
				Log.warning("Soft rate limit is too strict - it would have blocked this successful exposure detection", log: .riskDetection, logger: logger)
```

Reply:
typo fixed